src/AutorestSwift/Code Generator/Manager.swift

```diff
@@ -33,56 +33,68 @@ struct StringDecodeKeys: ExtensionStringDecodeKeys {
 
 /// Handles the configuration and orchestrations of the code generation process
 class Manager {
-    internal enum InvocationMode {
-        case autorest
-        case commandLine
-    }
+    static var shared = Manager()
 
     // MARK: Properties
 
-    let inputString: String
-
-    let destinationRootUrl: URL
+    var configured: Bool
+    var config: ManagerConfig
+    var inputString: String
+    var destinationRootUrl: URL!
     var packageUrl: URL?
 
-    let mode: InvocationMode
+    // MARK: Initializers
 
-    static var packageName: String?
+    init() {
+        self.inputString = ""
+        self.config = ManagerConfig()
+        self.configured = false
+        self.packageUrl = nil
+        self.destinationRootUrl = nil
+    }
 
-    // MARK: Initializers
+    // MARK: Methods
 
     /// Initialize with an input file URL (i.e. running locally).
-    init(withInputUrl input: URL, destinationUrl dest: URL) throws {
-        self.mode = .commandLine
-
+    func configure(input: URL, destination: URL) throws {
+        guard configured == false else {
+            SharedLogger.warn("Manager is already configured.")
+            return
+        }
         let yamlString = try String(contentsOf: input)
-        self.inputString = yamlString
-        self.packageUrl = nil
+        inputString = yamlString
 
-        // TODO: Make this configurable
-        self.destinationRootUrl = dest.appendingPathComponent("generated").appendingPathComponent("sdk")
+        // FIXME: Make this configurable
+        let destUrl = destination.appendingPathComponent("generated").appendingPathComponent("sdk")
+        destinationRootUrl = destUrl
         do {
-            try destinationRootUrl.ensureExists()
+            try destUrl.ensureExists()
         } catch {
             SharedLogger.fail("\(error)")
         }
+        configured = true
     }
 
     /// Initialize with a raw YAML string (i.e. the way it is received from Autorest).
-    init(withString string: String, packageName: String? = nil) {
-        self.mode = .autorest
-        self.inputString = string
+    func configure(input: String, client: ChannelClient, sessionId: String, completion: @escaping () -> Void) {
+        guard configured == false else {
+            SharedLogger.warn("Manager is already configured.")
+            return
```

Comment:
Should this also call `fatalError` when configured is true, same as what you did on line 96

Reply:
If it's already configured, this just warns you and lets you continue with how it was previously configured. However, if you try to run autorest without configuring it, there's no possible way it will work, which is why that is a fatal error.